scanner/scanner.go

```diff
@@ -108,6 +117,100 @@ func (s *Scanner) ScanAndClean(opts ScanOptions) (*Context, error) {
 	return c, nil
 }
 
+func (s *Scanner) ExecuteWatch() error {
+	var err error
+	s.watcher, err = fsnotify.NewWatcher()
+	if err != nil {
+		log.Fatalf("error creating watcher: %v\n", err)
+	}
+	defer s.watcher.Close()
+
+	for _, dir := range s.musicDirs {
+		err := filepath.WalkDir(dir, func(absPath string, d fs.DirEntry, err error) error {
+			return s.watchCallback(dir, absPath, d, err)
+		})
+		if err != nil {
+			log.Printf("error watching directory tree: %v\n", err)
+		}
+	}
+
+	if _, err = s.ScanAndClean(ScanOptions{}); err != nil {
+		log.Printf("error scanning: %v", err)
+	}
+
+	for {
+		select {
+		case event := <-s.watcher.Events:
+			doScan := event.Op&(fsnotify.Create|fsnotify.Write) != 0
+			if doScan && s.StartScanning() {
```

Comment:
what if we inverted this and undented this a bit
```suggestion
			if !doScan || !s.StartScanning() {
			    break
			}
			// ... stuff
```

Reply:
I can do that but looks like your suggested change does not include the dedent or the removal of the closing brace, correct?

Reply:
OK I committed the change.